src/Orleans/AssemblyLoader/AssemblyCatalog.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Orleans.Runtime
+{
+    public class AssemblyCatalog : IAssemblyCatalog
```

Comment:
There might be changes to the catalog interface too, but I'd be surprised if they couldn't be backawards compatible if needed.